rcl/include/rcl/context.h

```diff
@@ -240,6 +240,31 @@ RCL_WARN_UNUSED
 rcl_context_instance_id_t
 rcl_context_get_instance_id(rcl_context_t * context);
 
+/// Returns the context domain id.
+/**
+ * The given context must be non-`NULL` and be valid.
```

Comment:
@ivanpauno consider clarifying what valid means in this context. Initialized and not finalized perhaps? Or initialized and not shutdown? Also, consider using `/pre`.

Reply:
> initialized and not finalized perhaps? Or initialized and not shutdown?

It means the second one.

> Also, consider using /pre.

This is not a precondition, because the function handles those cases correctly returning an appropriate error code.

Reply:
See https://github.com/ros2/rcl/pull/718/commits/85708bed10368e0ddc973c1c44fce2b8d7f513a7.

I'm not sure how to rephrase uninitialized to be more clear.
e.g. see [this other docblock](https://github.com/ros2/rcl/blob/1a7f737fa9484b93d8c3052d37ad3ac55c921f2f/rcl/include/rcl/context.h#L217-L219), that's really confusing.

Reply:
Agreed about not being a `/pre`, my bad. That other docblock is insane.